### PR TITLE
Port to Python 3

### PR DESCRIPTION
**Pull request by andijakl**
Ported the scripts to run on Python 3 (after all, Python 2 won't be maintained anymore after 2020). For the main Arlo.py script, this includes only that the Queue has been renamed to queue.

Example script: Writing the byte stream is different in Python 3 (no more implicit conversions between byte arrays and strings). I also changed the file name to be human-readable and to include the unique ID to prevent file name clashes for multiple cameras. Plus, the print statement now needs to have the string argument surrounded by ().

Also added the example download script in the readme as directly executable script file to the repository for easier testing.

More background info about the changes: https://www.andreasjakl.com/using-netgear-arlo-security-cameras-for-periodic-recording/

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -41,35 +48,36 @@ try:
...
 
+		videofilename = datetime.datetime.fromtimestamp(int(recording['name'])//1000).strftime('%Y-%m-%d %H-%M-%S') + ' ' + recording['uniqueId']
 		##
 		# The videos produced by Arlo are pretty small, even in their longest, best quality settings,
 		# but you should probably prefer the chunked stream (see below). 
 		###    
 		#    # Download the whole video into memory as a single chunk.
 		#    video = arlo.GetRecording(recording['presignedContentUrl'])
-		#	 with open(recording['name']+'.mp4', 'w') as f:
+		#	 with open(videofilename+'.mp4', 'w') as f:
```
Comment (jeffreydwalter) [inline README.md]: Minor tweak, but maybe we should append .mp4 to videofilename on line 51 where you assign videofilename so we don't have to keep appending it.

File: arlo-download.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+	# Get all of the recordings for a date range.
+	library = arlo.GetLibrary(seven_days_ago, today)
+
+	# Iterate through the recordings in the library.
+	for recording in library:
+
+		# Get video as a chunked stream; this function returns a generator.
+		stream = arlo.StreamRecording(recording['presignedContentUrl'])
+		videofilename = datetime.datetime.fromtimestamp(int(recording['name'])//1000).strftime('%Y-%m-%d %H-%M-%S') + ' ' + recording['uniqueId']
```
Comment (jeffreydwalter) [inline arlo-download.py]: Same as in the README. Let's append the .mp4 to videofilename, so we can stop appending it all over the place below.

**Review by jeffreydwalter — CHANGES_REQUESTED** [review]
Other than my two comments (which are about the same thing), everything looks good. Thanks for contributing!

File: arlo-download.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
+from datetime import timedelta, date
```
Comment (jeffreydwalter) [inline arlo-download.py]: Might be good to move this script to an examples folder.

**Review by jeffreydwalter — CHANGES_REQUESTED** [review]

**Comment by jeffreydwalter** [comment]
Are these changes breaking change for Python 2? If so, perhaps we should include an Arlo.py for both versions of Python or better would be to make the changes required to support both in a single script?

**Comment by andijakl** [comment]
Thanks for your comments! I've addressed these and made another commit.

I now kept the origional Python 2 file as "Arlo-Python2.py". Not sure if it'd be possible to have a version that works on both, as the "queue" package was renamed and the division import from future is no longer needed. So out of the box, the new Python 3 version wouldn't run in 2. But I feel it's better to have the Python 3 version as default by now.

For the download example, I'm assuming that the udpated version should also work fine with Python 2. The now mandatory paranthesis around the text of the print statement have been allowed in Python 2 as well. The additional buffer statement for writing the byte array to the file has been available in Python 2 already, so I think the code example should work fine also with the old Python version!

**Comment by jeffreydwalter** [comment]
Instead of having two versions would it be possible to do something like this:
```
if sys.version[0] == '2':
    import Queue as queue
else:
    import queue as queue
```
?

**Comment by andijakl** [comment]
Good idea - changed that in the Arlo.py script - still works fine (with Python 3). Removed the other extra copy of the script for Python 2.

**Review by jeffreydwalter — APPROVED** [review]
Looks great! Thanks for contributing!

**Pull request merged**